docs/cli/ilab-model-serve.md

```diff
@@ -0,0 +1,103 @@
+
+`ilab model serve` has a subcommand for each backend it can serve models with. As of this draft, the two backends `ilab` would serve with are `llama-cpp` and `vllm`.
+
+This would lead to the commands:
+
+- `ilab model serve llama-cpp`
+- `ilab model serve vllm`
+
+There are specific flags for `ilab model serve` that would apply to all backends. These can be viewed by running `ilab model serve --help`. 
+For the specific subcommands, each has their own flags that can be viewed via `ilab model serve <BACKEND> --help`.
+This ensures there is no mix up of flags between backends.
+
+The following is an overview for the flags of `ilab model serve`:
+
+```console
+ilab model serve
+|
+|_______ --model
+|
+|_______ (other backend agnostic flags)
+|
+|_______ llama-cpp
+|        |
+|        |_______ --num-gpu-layers
+|        |
+|        |_______ --max-ctx-size
+|        |
+|        |_______ (other llama-cpp specific args)
+|
+|_______ vllm
+|        |
+|        |_______ (popular vllm-specific specific args)
+|        |
+|        |_______ --vllm-args
+```
+
+## Command Examples
+
+### Bare-bones but model specific command
+
+```shell
+ilab model serve --model <PATH> 
+```
+
+- Serves the model at `<PATH>`.
+- If the `<PATH>` is the path for a model that can be run by `llama-cpp` then `llama-cpp` is automatically used as the model serving backend.
+- If the `<PATH>` is the path for a model that can be run by `vllm` then `vllm` is automatically used as the model serving backend.
+- If the model at `<PATH>` can be run by either backend, then a default backend defined in the config will be used as the model serving backend.
+
+### Bare-bones command
+
+```shell
+ilab model serve
+```
+
+- This command has the same behavior as the one above but the `--model` is whatever the default model path is in the config. This is the existing behavior of `ilab serve` today.
+
+### Llama-cpp backend specific commands
+
+```shell
+ilab model serve --model <PATH> llama-cpp --num-gpu-layers 4 --max-ctx-size 1024
+```
+
+- This command serves a model with `llama-cpp`.
+- If the model provided is not able to be served by llama-cpp, this command would error out and suggest an alternate backend to use.
+- The existing flags to `ilab serve` (besides `--model-path` & `--log-file`) are now specific to the llama-cpp backend.
+
+### vllm backend specific commands
+
+```shell
+ilab model serve --model <PATH> vllm --chat-template <PATH> --vllm-args "<MORE VLLM SPECIFIC ARGS>"
+```
+
+- This command serves a model with `vllm`.
+- If the path provided is not able to be served by `vllm`, this command would error out and suggest an alternate backend to use.
+- There are [dozens](https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html#command-line-arguments-for-the-server) of flags for vllm. Whichever arguments the community deems the most important to include, will be added as flags to `ilab model serve vllm`.
```

Comment:
It’s probably easier to just have a pass through mechanism. I don’t see much value in replicating their config options in our own.

Reply:
I see the next line proposes that already… I’d prefer stating that pass through is the default answer unless we have a really good reason otherwise

Reply:
+1 thats where I land. Ideally generic shared options that are intuitive and then passthrough so we don't have to keep a mirror in sync.